game/overlord/iso.cpp

```diff
@@ -793,6 +823,120 @@ u32 ProcessVAGData(IsoMessage* _cmd, IsoBufferHeader* buffer_header) {
 // TODO - UpdatePlayPos
 // TODO - CheckVAGStreamProgress
 
+u32 GetPlayPos();
+void UpdatePlayPos();
+void StopVAG(VagCommand* cmd);
+void PauseVAG(VagCommand* cmd);
+void UnpauseVAG(VagCommand* cmd);
+void CalculateVAGVolumes(u32 a1, u32 a2, u32 a3, int* a4, int* a5);
+void SetVAGVol();
+
+u32 GetPlayPos() {
+  u32 addr;
+  while (1) {
+    addr = sceSdGetAddr(gVoice | 0x2240);
+    u32 addr1 = sceSdGetAddr(gVoice | 0x2240);
+    u32 addr2 = sceSdGetAddr(gVoice | 0x2240);
+    if (addr == addr1 || addr == addr2)
+      break;
+    if (addr1 == addr2) {
+      addr = addr1;
+      break;
+    }
+  }
+  u32 play_pos = addr - gStreamSRAM;
+  if (play_pos > BUFFER_PAGE_SIZE - 1)
+    return 0xFFFFFFFF;
+  else
+    return play_pos;
+}
+
+void UpdatePlayPos() {
+  u32 play_pos;
+  if (gPlaying) {
+    play_pos = GetPlayPos();
+    if (play_pos == 0xFFFFFFFF) {
+      if (gLastVAGHalf) {
+        play_pos = BUFFER_PAGE_SIZE;
+      } else {
+        play_pos = STR_BUFFER_DATA_SIZE;
+      }
+    } else {
+      gLastVAGHalf = play_pos >= STR_BUFFER_DATA_SIZE;
+    }
+    if (play_pos >= gPlayPos) {
+      gRealVAGClockS += play_pos - gPlayPos;
+    } else {
+      gRealVAGClockS += play_pos + BUFFER_PAGE_SIZE - gPlayPos;
+    }
+    if (!gSampleRate) {
+      lg::error("[OVERLORD] gSampleRate is invalid!");
+    }
+    gRealVAGClock = 4 * (0x1C00 * (gRealVAGClockS >> 4) / gSampleRate);
+    gPlayPos = play_pos;
+  }
+}
+
+// TODO
+void StopVAG(VagCommand* cmd) {
+  gPlaying = 0;
+  PauseVAG(cmd);
+  // snd_KeyOffVoiceRaw(gVoice & 1, gVoice >> 1);
+  gFakeVAGClockRunning = 0;
+  gRealVAGClockRunning = 0;
+  gVAGId = 1;
+}
+
+// TODO
+void PauseVAG(VagCommand* cmd) {
+  gFakeVAGClockPaused = 1;
+  // 0x3c = paused?
+  if (cmd->field_0x3c) {
+    cmd->field_0x3c = 1;
+    if (cmd->field_0x38) {
+      // sceSdProcBatch();
+    }
+  }
+}
+
+// TODO
+void UnpauseVAG(VagCommand* cmd) {
+  gFakeVAGClockPaused = 0;
+}
+
+// TODO - not sure what any of these are yet
+void CalculateVAGVolumes(u32 a1, u32 a2, u32 a3, int* a4, int* a5) {
+  if (a2) {
+    u32 falloff_vol = 0;  // CalculateFalloffVolume(a3, (a1 * gDialogVolume) >> 10, 1, 10, 50);
+    short angle = 0;      // CalculateAngle(a3);
+    *a4 = (angle * falloff_vol) >> 10;
+    *a5 = (angle * falloff_vol) >> 10;
```

Comment:
my guess is this is calculating the volume for the left/right audio channels, based on the position of the camera/cutscene audio sorce.